networkx/algorithms/community/tests/test_asyn_lpa.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python
+from nose.tools import assert_equal
+from networkx import asyn_lpa, Graph
+
+
+def test_empty_graph():
+    # empty graph
+    test = Graph()
+
+    # ground truth
+    ground_truth = set()
+
+    communities = asyn_lpa.asyn_lpa_communities(test)
+    result = set()
+    for c in communities:
+        result.add(frozenset(c))
```

Comment:
Simpler: `result = {frozenset(c) for c in communities}`, and everywhere this pattern appears below.